### ISSUE TITLE
Purpose of URLParameterEncoding.default?

### ISSUE THREAD

**Issue by scottrhoyt** [Alamofire/Alamofire#1872]
I was wondering what the purpose of `URLParameterEncoding.default` is. It isn't a singleton allowing for sharing a single instance of a `URLParameterEncoding`. Likewise, initializing a `URLParameterEncoding` with default options seems to be better served by just calling the initializer with no parameters. This pattern also applies to `JSONEncoding` and `PropertyListEncoding`.

**Comment by jshier** [comment]
@cnoon designed this API, but I believe it was to match the previous `enum` implementation and assist people in transitioning to the new version. It also establishes the general pattern of convenience static properties for preset values. You're right, these could (and likely should) be singleton values, rather than generating a new one every time, but that rather minor refinement can be made at any time. @cnoon, anything to add?

**Comment by scottrhoyt** [comment]
Thanks for the response, @jshier!

I don't think the performance penalty of creating a `URLParameterEncoding` is too big of a deal. It more just caught me off guard when I wrote a unit test that essentially amounted to `URLParameterEncoding.default === URLParameterEncoding.default` and it failed. I had mistakenly assumed it was a singleton. When I dug into it I was curious what the reasoning was because it seemed like `URLParameterEncoding()` would be more idiomatic to create a new instance with default values. I figured it might be a legacy purpose. Thanks for satisfying my curiosity!
